code/game/gamemodes/miniantags/witchcraft/witch_datum.dm

```diff
@@ -0,0 +1,21 @@
+//The datum used in mob minds to control the mob's knowledge/affinity in witchcraft.
+
+#define AFFINITY_EARTH 1
+#define AFFINITY_FIRE 2
+#define AFFINITY_WATER 3
+#define AFFINITY_AIR 4
+#define AFFINITY_MAGIC 5
+
+var/list/witches = list() //All witches in the game
```

Comment:
That or https://github.com/tgstation/-tg-station/blob/master/code/_globalvars/lists/mobs.dm